frontends/precinct-scanner/src/components/replace_ballot_bag_modal.tsx

```diff
@@ -0,0 +1,110 @@
+import React, { useEffect, useState } from 'react';
+import { Button, Text, Modal } from '@votingworks/ui';
+import { throwIllegalValue } from '@votingworks/utils';
+import { ScannedBallotCount } from './scanned_ballot_count';
+import { CenteredLargeProse } from './layout';
+import { BALLOT_BAG_CAPACITY } from '../config/globals';
+
+interface Props {
+  scannedBallotCount: number;
+  pollWorkerAuthenticated: boolean;
+  onComplete: VoidFunction;
+}
+
+type ModalState = 'no_auth' | 'confirming' | 'confirmed';
+
+export function ReplaceBallotBagModal({
+  scannedBallotCount,
+  pollWorkerAuthenticated,
+  onComplete,
+}: Props): JSX.Element {
+  const [modalState, setModalState] = useState<ModalState>('no_auth');
+
+  useEffect(() => {
+    if (modalState === 'no_auth' && pollWorkerAuthenticated) {
+      setModalState('confirming');
+    } else if (modalState === 'confirming' && !pollWorkerAuthenticated) {
+      setModalState('no_auth');
+    } else if (modalState === 'confirmed' && !pollWorkerAuthenticated) {
+      onComplete();
+    }
+  }, [modalState, onComplete, pollWorkerAuthenticated]);
+
+  const mainContent = (() => {
+    switch (modalState) {
+      case 'no_auth':
+        return (
+          <CenteredLargeProse>
+            <h1>The Ballot Bag is Full</h1>
+            <p>
+              A Poll Worker must replace the full ballot bag with an empty
+              ballot bag.
+            </p>
+            <p>Insert a Poll Worker Card after the ballot bag is replaced.</p>
+          </CenteredLargeProse>
+        );
+      case 'confirming':
+        return (
+          <CenteredLargeProse>
+            <h1>Ready to Resume Voting?</h1>
+            <p>
+              Has the full ballot bag has been replaced with an empty ballot
+              bag?
+            </p>
+            <p>
+              <Button primary onPress={() => setModalState('confirmed')}>
+                Yes, Resume Voting
+              </Button>
+            </p>
+            <Text small italic>
+              Remove card if you’re not ready to resume voting.
```

Comment:
This instruction confused me when I watched the video. If I were a poll worker and I wasn't ready to resume voting, wouldn't I just not take any action? Why do they need to remove the card?

Reply:
I want the poll worker to know that they can remove the card in the case that they need to before replacing the bag. I want the poll worker to have confidence and know what their options are. 

It would be odd if a poll worker inserted their card, then left their card in the machine and walked away to get a new ballot bag.